cmd/aergocli/cmd/staking.go

```diff
@@ -16,55 +18,56 @@ import (
 var stakingCmd = &cobra.Command{
 	Use:   "staking",
 	Short: "Staking balance to aergo system",
-	Run:   execStaking,
+	RunE:  execStaking,
 }
 
-func execStaking(cmd *cobra.Command, args []string) {
-	sendStaking(cmd, true)
+func execStaking(cmd *cobra.Command, args []string) error {
+	return sendStaking(cmd, true)
 }
 
 var unstakingCmd = &cobra.Command{
 	Use:   "unstaking",
 	Short: "Unstaking balance from aergo system",
-	Run:   execUnstaking,
+	RunE:  execUnstaking,
 }
 
-func execUnstaking(cmd *cobra.Command, args []string) {
-	sendStaking(cmd, false)
+func execUnstaking(cmd *cobra.Command, args []string) error {
+	return sendStaking(cmd, false)
 }
 
-func sendStaking(cmd *cobra.Command, s bool) {
+func sendStaking(cmd *cobra.Command, s bool) error {
 	account, err := types.DecodeAddress(address)
 	if err != nil {
-		cmd.Printf("Failed: (%s) %s\n", address, err.Error())
-		return
+		return errors.New("failed to parse --address flag (" + address + ")\n" + err.Error())
 	}
 	payload := make([]byte, 1)
 	if s {
 		payload[0] = 's'
 	} else {
 		payload[0] = 'u'
 	}
-	if amount < types.StakingMinimum {
-		cmd.Printf("Failed: minimum staking value is %d\n", types.StakingMinimum)
-		return
+	amountBigInt, ok := new(big.Int).SetString(amount, 10)
+	if !ok {
+		return errors.New("failed to parse --amount flag\n" + err.Error())
+	}
+	if amountBigInt.Cmp(types.StakingMinimum) < 0 {
+		return errors.New("Failed: minimum staking value is " + types.StakingMinimum.String())
 	}
 
 	tx := &types.Tx{
 		Body: &types.TxBody{
 			Account:   account,
 			Recipient: []byte(types.AergoSystem),
-			Amount:    amount,
-			Price:     0,
+			Amount:    amountBigInt.Bytes(),
```

Comment:
Just a question, what is the point of calling Bytes()? Is it to create a copy?

Reply:
yes it's copy :(

https://github.com/golang/go/blob/master/src/math/big/int.go#L440